### PR TITLE
feat(sandbox): example of pubnub as a provider in notification service

### LINKED ISSUE #208: Need an example for pubnub as a provider in Notification service
I went through the examples and found that we presently do not have pubnub as a provider in any of our examples.
therefore, it would be resourceful to have an example with pubnub as a provider.

While creating the implementation I would request you to consider, Access management for channel subscriptions. posting messages to channels/channel groups.
for permission flow please consider https://www.pubnub.com/docs/security/access-control#initialize-with-secretkey

it would be an added advatage if we can also have a built in capability of monitoring presence at the service end where we use the pubnub webhooks to monitor users presence,
This might be a SRP breach but putting it here would kind of wrap the pubnub implementation at one place and accelerate the development.

### PR DESCRIPTION
**Pull request by yeshamavani**
MIGRATION CHANGE:
migration-20210716062747- notification migration
migration-20210501132806- notification migration

gh-208

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #208

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [x] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/core/locales/en.json [hunk 1]
```diff
@@ -16,5 +16,11 @@
 	"ENOENT": " no such file or directory, open '/home/samarpan/projects/loopback/loopback4-microservice-catalog/services/authentication-service/public/index.html'",
 	"The key 'sf.clientAuthentication.currentClient' is not bound to any value in context RequestContext-yWsyEvu3TeGuly75aFbRdg-13 (context": " RequestContext-yWsyEvu3TeGuly75aFbRdg-13, binding: sf.clientAuthentication.currentClient, resolutionPath: controllers.LoginController --> @LoginController.constructor[0])",
 	"Client Invalid": "Client Invalid",
-	"host is required": "host is required"
+	"host is required": "host is required",
+	"Cannot read property 'toLowerCase' of undefined": "Cannot read property 'toLowerCase' of undefined",
+	"The `users` instance is not valid. Details": " `firstName` can't be blank (value: undefined).",
```
Comment (akshatdubeysf) [inline packages/core/locales/en.json:21]: ??

File: sandbox/pubnub-example/src/application.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+require('dotenv').config();
+
+export class PubnubExampleApplication extends BootMixin(
+  ServiceMixin(RepositoryMixin(RestApplication)),
+) {
+  constructor(options: ApplicationConfig = {}) {
+    super(options);
+
+    // Set up the custom sequence
+    this.sequence(MySequence);
```
Comment (akshatdubeysf) [inline sandbox/pubnub-example/src/application.ts:35]: should only use the sequence from notification service.

**Review by akshatdubeysf — APPROVED** [review]

File: sandbox/pubnub-example/src/application.ts (hunk 2 above)
Comment (yeshamavani) [reply sandbox/pubnub-example/src/application.ts]: Done the required changes

File: sandbox/pubnub-example/src/application.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      path: '/explorer',
+    });
+    this.component(RestExplorerComponent);
+    this.component(NotificationServiceComponent);
+
+    this.bind(NotifServiceBindings.Config).to({
+      useCustomEmailProvider: false,
+      useCustomSMSProvider: false,
+      useCustomPushProvider: true,
+      useCustomSequence: true,
```
Comment (samarpan-b) [inline sandbox/pubnub-example/src/application.ts:47]: why custom sequence true ?

File: sandbox/pubnub-example/src/application.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    });
+
+    this.bind(PubnubBindings.Config).to({
```
Comment (samarpan-b) [inline sandbox/pubnub-example/src/application.ts:50]: can you put some comments as docs here about each config you are providing ? also dont hard code these values.

File: sandbox/pubnub-example/migrations/sqls/20210716062747-init-up.sql [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+	modified_on          timestamptz DEFAULT CURRENT_TIMESTAMP NOT NULL ,
+	id                   uuid DEFAULT md5(random()::text || clock_timestamp()::text)::uuid NOT NULL ,
+	notification_id      uuid  NOT NULL ,
+	user_id              uuid  NOT NULL ,
+	is_read              boolean DEFAULT false  ,
+	action_meta          text   ,
+	CONSTRAINT notification_users_pkey PRIMARY KEY ( id )
+ ) ;
+
+CREATE  TABLE main.notifications (
```
Comment (samarpan-b) [inline sandbox/pubnub-example/migrations/sqls/20210716062747-init-up.sql:20]: dont we need channels table ?

**Review by samarpan-b — CHANGES_REQUESTED** [review]

File: sandbox/pubnub-example/src/application.ts (hunk 3 above)
Comment (yeshamavani) [reply sandbox/pubnub-example/src/application.ts:47]: that needs to be changed, missed that

File: sandbox/pubnub-example/migrations/sqls/20210716062747-init-up.sql (hunk 5 above)
Comment (yeshamavani) [reply sandbox/pubnub-example/migrations/sqls/20210716062747-init-up.sql:20]: no i dont think so we need channel table

File: README.md [hunk 6]
```diff
@@ -1,57 +1,61 @@
 # Sourceloop
 
-[![LoopBack](https://github.com/strongloop/loopback-next/raw/master/docs/site/imgs/branding/Powered-by-LoopBack-Badge-(blue)-@2x.png)](http://loopback.io/)
+[![LoopBack](<https://github.com/strongloop/loopback-next/raw/master/docs/site/imgs/branding/Powered-by-LoopBack-Badge-(blue)-@2x.png>)](http://loopback.io/)
```
Comment (samarpan-b) [inline README.md:3]: revert this

File: sandbox/pubnub-example/src/application.ts (hunk 3 above)
Comment (yeshamavani) [reply sandbox/pubnub-example/src/application.ts]: done

File: sandbox/pubnub-example/migrations/sqls/20210716062747-init-up.sql (hunk 5 above)
Comment (yeshamavani) [reply sandbox/pubnub-example/migrations/sqls/20210716062747-init-up.sql:20]: not needed

File: README.md (hunk 6 above)
Comment (yeshamavani) [reply README.md]: these are because of rebasing I haven't added any changes to this

File: sandbox/pubnub-example/src/application.ts (hunk 4 above)
Comment (yeshamavani) [reply sandbox/pubnub-example/src/application.ts:60]: added comments  for config

**Review by akshatdubeysf — APPROVED** [review]

**Pull request merged**